app/controllers/users_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :update, :destroy]
+
+  def index
+    users = authorize User.where(admin: false)
```

Comment:
Acho que fica melhor gerar a query com a policy...

``` ruby
# user_policy.rb

class PostPolicy < ApplicationPolicy
  class Scope < Scope
      def resolve
        if user.admin?
          scope.all
        else
          scope.where(published: true)
        end
      end
    end
end

# models/user.rb

class User
  scope :members, -> { where(admin: false) } # só um exemplo do nome, members pode n ser o mais adequado aqui
end

# controllers/users_controller.rb

class UsersControllers < ApplicationController
  def index
    @users = policy_scope(User).members
  end
end
```

Reply:
Acho que nesse caso faz mais sentido usar o `authorize` dado que o objetivo é lançar um exception caso o usuário que não for admin tentar bater nesse endpoint. Posteriormente poderemos usar `scope` pro caso de um `Contributor` com permissão de `owner` em uma `Company` poder ver seus funcionários.